fem/fe/fe_pyramid.cpp

```diff
@@ -0,0 +1,1584 @@
+// Copyright (c) 2010-2025, Lawrence Livermore National Security, LLC. Produced
+// at the Lawrence Livermore National Laboratory. All Rights reserved. See files
+// LICENSE and NOTICE for details. LLNL-CODE-806117.
+//
+// This file is part of the MFEM library. For more information and source code
+// availability visit https://mfem.org.
+//
+// MFEM is free software; you can redistribute it and/or modify it under the
+// terms of the BSD-3 license. We welcome feedback and contributions, see file
+// CONTRIBUTING.md for details.
+
+// Finite Element classes on Pyramid shaped elements
+
+#include "fe_pyramid.hpp"
+
+namespace mfem
+{
+
+using namespace std;
+
+Vector FuentesPyramid::grad_lam1(real_t x, real_t y, real_t z)
+{
+   return Vector({CheckZ(z) ? - (1.0 - y - z) / (1.0 - z) : -0.5,
```

Comment:
Where do the values at z = 1.0 come from?

If we take y = \alpha (1.0 - z), then we get \lim_{z\to \infty}  -(1.0 - y - z) / (1.0 - z) =  \alpha - 1.0 

I suppose you are setting \alpha = 0.5?